### PR TITLE
[Refactor] refact warp server to Axum

### PR DESCRIPTION
**Pull request by ZhiHanZ**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

first stage for https://github.com/datafuselabs/datafuse/issues/1256
secondly we should remove warp dependencies in metric server

## Changelog

- Improvement

## Related Issues

Fixes #issue

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

**Comment by BohuTANG** [comment]
Conflicts need to solve

File: query/src/api/http/v1/hello.rs [hunk 1]
```diff
@@ -12,12 +12,68 @@
...
 
-pub fn hello_handler(
-    cfg: Config,
-) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
-    warp::path!("v1" / "hello").map(move || format!("{:?}", cfg))
+pub async fn hello_handler(cfg: Extension<Config>) -> String {
+    format!("{:?}", cfg)
+}
+
+#[tokio::test]
```
Comment (BohuTANG) [inline query/src/api/http/v1/hello.rs:24]: The test to a separate test file is better?

**Review by BohuTANG — CHANGES_REQUESTED** [review]
Neat, need some changes

File: query/src/api/http/v1/hello.rs (hunk 1 above)
Comment (ZhiHanZ) [reply query/src/api/http/v1/hello.rs]: separated

**Comment by ZhiHanZ** [comment]
> Conflicts need to solve

changed

File: query/src/api/http/v1/config.rs [hunk 2]
```diff
@@ -12,12 +12,10 @@
...
+use axum::extract::Extension;
 
 use crate::configs::Config;
 
-pub fn config_handler(
-    cfg: Config,
-) -> impl Filter<Extract = impl warp::Reply, Error = warp::Rejection> + Clone {
-    warp::path!("v1" / "configs").map(move || format!("{:?}", cfg))
+pub async fn config_handler(cfg: Extension<Config>) -> String {
+    format!("{:?}", cfg)
```
Comment (BohuTANG) [inline query/src/api/http/v1/config.rs:20]: cfg is Extension<Config> type, here we should format cfg.0

**Review by BohuTANG — APPROVED** [review]
LGTM
Thank you @ZhiHanZ

**Pull request merged**